src/network/network_content.cpp

```diff
@@ -607,12 +607,18 @@ void ClientNetworkContentSocketHandler::OnReceiveData(const char *data, size_t l
 	assert(data == nullptr || length != 0);
 
 	/* Ignore any latent data coming from a connection we closed. */
-	if (this->http_response_index == -2) return;
+	if (this->http_response_index == -2) {
+		if (data != nullptr) {
+			free(data);
+		}
```

Comment:
`free` accepts `nullptr` to reduce these kinds of special cases needing to be written.
```suggestion
		free(data);
```

Reply:
Old habit. We once had a platform where that caused a crash. It didn't follow POSIX all that well :P

Reply:
Using `std::unique_ptr<char[]>` instead of `const char *` in the signature of `OnReceiveData` and in `class HTTPThreadSafeCallback` may remove the need to have these manual calls to free at all.

Reply:
Yeah, did consider that, but the WinHTTP backend needs a `char *` pointer to fill; so that gets a bit tricky, especially as WinHTTP wants it in a bit annoying way. So the only way I saw to do a `unique_ptr` was to allocate yet another buffer, and copy it over. Which felt a bit meh at that point.

Reply:
You don't have to construct the `new []char` and the `unique_ptr` which becomes responsible for it at the same time.
You could do: `char *buffer = size == 0 ? nullptr : new char[size];` in WINHTTP_CALLBACK_STATUS_DATA_AVAILABLE.
Then construct your `std::unique_ptr<char[]>(static_cast<char *>(info))` in WINHTTP_CALLBACK_STATUS_READ_COMPLETE.

Reply:
I think I did this correct; at least, it works as expected :P